nodebuilder/p2p/flags.go

```diff
@@ -50,3 +61,52 @@ func ParseFlags(
 	}
 	return nil
 }
+
+// ParseNetwork tries to parse the network from the flags and environment,
+// and returns either the parsed network or the build's default network
+func ParseNetwork(cmd *cobra.Command) (Network, error) {
+	parsedNetwork := cmd.Flag(nodeNetworkFlag).Value.String()
+	// no network set through the flags, so check if there is an override in the env
+	if parsedNetwork == "" {
+		return ParseNetworkFromEnv()
+	}
+	return Network(parsedNetwork), nil
+}
+
+// ParseNetworkFromEnv tries to parse the network from the environment,
+// and returns either the parsed network or the build's default network
+func ParseNetworkFromEnv() (Network, error) {
+	network := DefaultNetwork
```

Comment:
Hmm, not so sure about this one. If the env doesn't find a network in its environment then, what should it return, nil?

Then we need to go from 
```go
func ParseNetwork(cmd *cobra.Command) (Network, error) {
	parsedNetwork := cmd.Flag(nodeNetworkFlag).Value.String()
	// no network set through the flags, so check if there is an override in the env
	if parsedNetwork == "" {
		return parseNetworkFromEnv()
	}
	return Network(parsedNetwork), nil
}
```

to something like 
```go
func ParseNetwork(cmd *cobra.Command) (Network, error) {
	parsedNetwork := cmd.Flag(nodeNetworkFlag).Value.String()
	// no network set through the flags, so check if there is an override in the env
	if parsedNetwork == "" {
		network, err := parseNetworkFromEnv()
		if network != nil {
		        return DefaultNetwork, err
		}
		return network, err
	}
	return Network(parsedNetwork), nil
}
```

which is more awkward imo. Does this snippet change your mind? otherwise ill just implement as you suggest

Reply:
I don't think it's more awkward -- it's more explicit then what `parseNetworkFromEnv` is doing

`parseNetworkFromEnv` can then return (Network, bool) and you can check envNetwork, ok := ....; if ok {

Reply:
done, but not with (Network, bool)